cli/tools/repl.rs

```diff
@@ -492,6 +531,34 @@ impl ReplSession {
     Ok(value.to_string())
   }
 
+  async fn evaluate_ts_expression(
+    &mut self,
+    expression: &str,
+  ) -> Result<Value, AnyError> {
+    let parsed_module =
+      crate::ast::parse("repl.ts", &expression, &crate::MediaType::TypeScript)?;
+
+    let transpiled_src = parsed_module
+      .transpile(&crate::ast::EmitOptions {
+        emit_metadata: false,
+        source_map: false,
+        inline_source_map: false,
+        imports_not_used_as_values: ImportsNotUsedAsValues::Preserve,
+        // JSX is not supported in the REPL
+        transform_jsx: false,
+        jsx_factory: "React.createElement".into(),
+        jsx_fragment_factory: "React.Fragment".into(),
+      })?
+      .0;
+
+    self
+      .evaluate_expression(&format!(
+        "'use strict'; void 0;\n{}",
+        transpiled_src
+      ))
+      .await
```

Comment:
A side effect of this is when the transformed AST gets printed the line numbers will no longer match.

For example, before (note line number is 2):

![image](https://user-images.githubusercontent.com/1609021/121712148-6bb15680-caa9-11eb-84d1-4b920f942331.png)

And after (now it's 5):

![image](https://user-images.githubusercontent.com/1609021/121712183-779d1880-caa9-11eb-9abd-04b10b24a946.png)

Is there a way we can fix this easily with the source map since swc can provides that? I'm not super familiar with this, but can look into it if someone believes it's worthwhile.

Reply:
> Is there a way we can fix this easily with the source map since swc can provides that? I'm not super familiar with this, but can look into it if someone believes it's worthwhile.

I don't think it would be that simple - we'd have to store related source map for the last evaluated expression and perform ad-hoc mapping on errors. I think we can skip it for the first pass?